application/backend/app/services/system_service.py

```diff
@@ -28,3 +31,50 @@ def get_cpu_usage(self) -> float:
             float: CPU usage in percentage
         """
         return self.process.cpu_percent(interval=None)
+
+    @staticmethod
+    def get_devices() -> list[DeviceInfo]:
+        """
+        Get available compute devices (CPU, Intel XPU, NVIDIA CUDA)
+
+        Returns:
+            list[DeviceInfo]: List of available devices
+        """
+        # CPU is always available
+        devices: list[DeviceInfo] = [DeviceInfo(type="cpu", name="CPU", memory=None, index=None)]
+
+        # Check for Intel XPU devices
+        try:
+            if torch.xpu.is_available():
+                for device_idx in range(torch.xpu.device_count()):
+                    dp = torch.xpu.get_device_properties(device_idx)
+                    devices.append(
+                        DeviceInfo(
+                            type="xpu",
+                            name=dp.name,
+                            memory=dp.total_memory,
+                            index=device_idx,
+                        )
+                    )
+        except (ImportError, AttributeError, RuntimeError):
+            # torch not available, or XPU not supported
+            pass
```

Comment:
The import error will never trigger here, the `torch.xpu` and `torch.cuda` modules always exist regardless of the platform; the `torch.xpu.is_available()` function checks if such modules are functioning or not.

I suggest to remove this `try...except` block entirely, the enclosed code should never raise an exception. In the unlikely case it does, it's still better to raise an exception that we can investigate rather than silently skipping.